exercises/04-exercise.py

```diff
@@ -0,0 +1,220 @@
+These are hardware level details.
+Use these to apply to the 'counts' column for an files etc
+'''
+fc = 7.62939453125e-05
+adc2counts = lambda x: ((int(x, 16) >> 8) - 0x40000) * fc \
+        if (int(x, 16) >> 8) > 0x1FFFF else (int(x, 16) >> 8)*fc
+enc2counts = lambda x: int(x) if int(x) <= 0 else -(int(x) ^ 0xffffff - 1)
+
+def raw_read_in(raw_files):
+	'''
+	the raw files to read in
+	Returns a list of raw files read in as arrays
+	'''
+	raw_files_read_in = []
+	index = 0
+	for file in raw_files:
+		readin_file = np.loadtxt(directory_string + "/" + raw_files[index])
+		raw_files_read_in.append(readin_file)
+		index += 1
+	return raw_files_read_in
+
+def files_in_dir(raw_files_read_in):
+	'''
+	returns the list of raw files read in, that is, the data for 
+	each file is stored in an index of a ndarray
+	returns a numpy array
+	'''
+	return np.asarray(raw_files_read_in)
+
+def print_size_files(file_read_in_dir, raw_files):
+	'''
+	prints how many rows and how many columns there are for 
+	each raw file read in
+	'''
+	for index in range(len(file_read_in_dir)):
+		print(str(raw_files[index]) + " - (Rows, Columns) - "+ str(file_read_in_dir[index].shape))
+
+def total_time(an_files_only, directory_string):
+	'''
+	adding the sum of the time_in_sec AND time_in_nanosec of each ROW to a list
+	once the list reaches the last index, the list is stored in another list that
+	will hold the sum of time, from each row, of x file
+	returns a list of lists, the lists in each index contain the 
+	totaltime of that specific file , 
+	e.g.: time2totaltime[0] = list of totaltime for each row from file associated with this specific index
+	'''
+	time = 0.0
+	temp_total_time = []
+	time2totaltime = []
+
+	for outer_index in range(len(an_files_only)):
+		read_an_file = np.loadtxt(directory_string + "/" + an_files_only[outer_index])
+		for inner_index in range(read_an_file.shape[0]): # less than number of rows in file
+			if inner_index != len(read_an_file): # if inner index != the length or num of elements of file
+				time_in_sec = read_an_file[inner_index][0] #gets time in sec
+				time_in_nanosec = read_an_file[inner_index][1]#gets time for nanosec
+				temp_total_time.append(time_in_sec + time_in_nanosec*1e-9)#sums to total time
+				inner_index += 1 #move to next row
+		time2totaltime.append(temp_total_time)
+		temp_total_time = []
+
+	return time2totaltime
+
+def sum_times(time2totaltime):
+	'''
+	for each raw file read in that has the two columns of time
+	summed up already, sum up the entire column of time2totaltime(list)
+	for each file read in
+	'''
+	sum_of_time = []
+	for time in time2totaltime:
+		sum_of_time.append(np.sum(time))
+
+	return sum_of_time
+
+def print_sum_of_time(sum_of_time):
+	'''
+	prints the total sum of the totaltime column for each file
+	'''
+	for i in range(len(sum_of_time)):
+		print('Sum of total time in {0} = {1}s'.format(an_files_only[i], sum_of_time[i]))
+	print('\n')
+
+
+def an_files_to_dict(an_files_only):
```

Comment:
I would recommend having a function `an_file_to_dict` . This function would then loop over this new function. Can you try that?